libs/sdk-ui-kit/src/Dialog/ShareDialog/ShareDialogBase/GranularPermissions/GranularPermissionItem.tsx

```diff
@@ -32,8 +31,10 @@ const GranularPermissionSelectItem: React.FC<IGranularPermissionItemProps> = ({
     const handleOnChange = useCallback(
         (permission: IGranularPermissionTypeItem) => {
             toggleDropdown();
-            handleSetSelectedPermission(permission.id);
-            onChange({ ...grantee, permissions: [permission.id] });
+            if (permission.id !== selectedPermission) {
+                handleSetSelectedPermission(permission.id);
+                onChange({ ...grantee, permissions: [permission.id] });
+            }
         },
         [grantee, toggleDropdown, handleSetSelectedPermission, onChange],
```

Comment:
`selectedPermission` is missing in dependencies

Reply:
fixed